### DISCUSSION TITLE
Write a new sector-agnostic core to DOLPHYN

### DISCUSSION THREAD

**Discussion by RuaridhMacd** [macroenergy/DOLPHYN#142] [Ideas]
Currently, DOLPHYN is structured as multiple other sectors build around the GenX electricity sector model. This creates many headaches for development and use, especially if trying to run optimizations which focus on other sectors.

Others (e.g. @Betristor) have raised the idea of rewriting DOLPHYN to make all of the core infrastructure sector-agnostic; e.g. creating an objective function, loading data, writing outputs; and then allow each module to be added or removed more easily. This could include having sectors represented by price timeseries rather than full optimizations. 

We should also keep decomposition methods in mind which allow us to solve each sector in parallel.

**Comment by dharik13** [comment]
This is an important functionality that we should consider prioritizing.
